docs/applications/containers/beginners-guide-to-LXD-reverse-proxy/index.md

```diff
@@ -0,0 +1,668 @@
+{{< /output >}}
+
+    There are three containers, all in the *RUNNING* state, and each has their own private IP address. Take note of the IP addresses (both IPv4 and IPv6) for the container `proxy`. You will need them when you configure the `proxy` container in a later section.
+
+    You have created the containers. In the following sections, you will setup the web server software in the `apache1` and `nginx1` containers. Then, you will setup the `proxy` container so that the web servers are accessible from the Internet.
+
+### Configuring the Apache Web Server Container
+
+When using a reverse proxy in front of a web server, the web server does not know the IP addresses of visitors. The web server only sees the IP address of the reverse proxy. However, each web server has a way to identify the real remote IP address of a visitor. For Apache, this is performed with the Remote IP Apache module. For the module to work, the reverse proxy must be configured to pass the information regarding the remote IP addresses.
+
+1.  Start a shell in the `apache1` container.
+
+        lxc exec apache1 -- sudo --user ubuntu --login
+
+1.  Update the package list in the `apache1` container.
+
+        sudo apt update
+
+1.  Install the package *apache2* in the container.
+
+        sudo apt install -y apache2
+
+1. Create the file `/etc/apache2/conf-available/remoteip.conf`.
+
+    {{< file "remoteip.conf" >}}
+RemoteIPHeader X-Real-IP
+RemoteIPTrustedProxy 10.10.10.28 fd42:67a4:b462:6ae2:216:3eff:fe00:252e
+{{</ file >}}
+
+    You can use the `nano` text editor, by running the command `sudo nano /etc/apache2/conf-available/remoteip.conf`. Note, these are the IP addresses of the `proxy` container shown earlier, for both IPv4 and IPv6. Replace these with the IPs from your `lxc list` output.
+
+    {{< note >}}
+Instead of specifying the IP addresses, you can also use the hostname, `proxy.lxd`. However, the RemoteIP Apache module is peculiar when using the hostname and will use only one of the two IP addresses (either IPv4 or IPv6), which means that the Apache web server will not know the real source IP address for some connections. By listing explicitly both IPv4 and IPv6 addresses, you can be certain that RemoteIP will successfully accept the source IP information from all connections of the reverse proxy.
+{{< /note >}}
+
+1. Enable the new `remoteip.conf` configuration.
+
+        sudo a2enconf remoteip
+
+    {{< output >}}
+Enabling conf remoteip.
+To activate the new configuration, you need to run:
+systemctl reload apache2
+{{< /output >}}
+
+1. Enable the `remoteip` Apache module.
+
+        sudo a2enmod remoteip
+
+    {{< output >}}
+Enabling module remoteip.
+To activate the new configuration, you need to run:
+systemctl restart apache2
+{{< /output >}}
+
+1. Edit the default web page for Apache to make a reference that it runs inside a LXD container.
```

Comment:
Here, and for the Nginx section as well, I moved this here and explicitly made this a step to complete instead of implying that it was done.